Jellyfin.Server.Implementations/Activity/ActivityManager.cs

```diff
@@ -0,0 +1,107 @@
+        public ActivityManager(JellyfinDbProvider provider)
+        {
+            _provider = provider;
+        }
+
+        /// <inheritdoc/>
+        public void Create(ActivityLog entry)
+        {
+            using var dbContext = _provider.GetConnection();
+            dbContext.ActivityLogs.Add(entry);
+            dbContext.SaveChanges();
+
+            EntryCreated?.Invoke(this, new GenericEventArgs<ActivityLogEntry>(ConvertToOldModel(entry)));
+        }
+
+        /// <inheritdoc/>
+        public async Task CreateAsync(ActivityLog entry)
+        {
+            using var dbContext = _provider.GetConnection();
+            await dbContext.ActivityLogs.AddAsync(entry);
+            await dbContext.SaveChangesAsync();
+
+            EntryCreated?.Invoke(this, new GenericEventArgs<ActivityLogEntry>(ConvertToOldModel(entry)));
+        }
+
+        /// <inheritdoc/>
+        public QueryResult<ActivityLogEntry> GetActivityLogEntries(DateTime? minDate, bool? hasUserId, int? startIndex, int? limit)
+        {
+            using var dbConnection = _provider.GetConnection();
+            var result = dbConnection.ActivityLogs.AsQueryable();
+
+            if (minDate.HasValue)
+            {
+                result = result.Where(entry => entry.DateCreated >= minDate.Value);
+            }
+
+            if (hasUserId.HasValue)
+            {
+                result = result.Where(entry => hasUserId.Value != Equals(entry.UserId, Guid.Empty));
+            }
+
+            if (startIndex.HasValue)
+            {
+                result = result.Where(entry => entry.Id >= startIndex.Value);
+            }
+
+            if (limit.HasValue && limit.Value > 0)
+            {
+                result = result.OrderByDescending(entry => entry.DateCreated).Take(limit.Value);
+            }
+
+            // This converts the objects from the new database model to the old for compatibility with the existing API.
+            var list = result.Select(entry => ConvertToOldModel(entry)).ToList();
+
+            return new QueryResult<ActivityLogEntry>()
+            {
+                Items = list
+            };
+        }
+
+        /// <inheritdoc/>
+        public QueryResult<ActivityLogEntry> GetActivityLogEntries(DateTime? minDate, int? startIndex, int? limit)
+        {
+            return GetActivityLogEntries(minDate, null, startIndex, limit);
+        }
+
+        private static ActivityLogEntry ConvertToOldModel(ActivityLog entry)
+        {
+            return new ActivityLogEntry
+            {
+                Id = entry.Id,
+                Name = entry.Name,
+                Overview = entry.Overview,
+                ShortOverview = entry.ShortOverview,
+                Type = entry.Type.ToString(),
+                ItemId = entry.ItemId,
+                UserId = entry.UserId,
+                Date = entry.DateCreated,
+                Severity = entry.LogSeverity
```

Comment:
What about `UserPrimaryImageTag`?

Reply:
As far as I can tell, that was completely unused. It was never stored in the database, and no code calls it. I don't think it should be, as that should be stored within the referenced User.

Reply:
Ah okay makes sense. Maybe let's just get rid of that property then? We might not want to change the object being returned from the API though, so maybe just add a `[Deprecated]` tag to it?